ibm/service/atracker/data_source_ibm_atracker_targets.go

```diff
@@ -311,55 +301,37 @@ func DataSourceIBMAtrackerTargetsTargetToMap(model *atrackerv2.Target) (map[stri
 		}
 		modelMap["cloudlogs_endpoint"] = []map[string]interface{}{cloudlogsEndpointMap}
 	}
-	if model.WriteStatus != nil {
-		writeStatusMap, err := DataSourceIBMAtrackerTargetsWriteStatusToMap(model.WriteStatus)
-		if err != nil {
-			return modelMap, err
-		}
-		modelMap["write_status"] = []map[string]interface{}{writeStatusMap}
-	}
-	if model.CreatedAt != nil {
-		modelMap["created_at"] = model.CreatedAt.String()
-	}
-	if model.UpdatedAt != nil {
-		modelMap["updated_at"] = model.UpdatedAt.String()
+	writeStatusMap, err := DataSourceIBMAtrackerTargetsWriteStatusToMap(model.WriteStatus)
+	if err != nil {
+		return modelMap, err
 	}
-	if model.APIVersion != nil {
-		modelMap["api_version"] = *model.APIVersion
+	modelMap["write_status"] = []map[string]interface{}{writeStatusMap}
+	modelMap["created_at"] = model.CreatedAt.String()
+	modelMap["updated_at"] = model.UpdatedAt.String()
+	if model.Message != nil {
+		modelMap["message"] = *model.Message
 	}
+	modelMap["api_version"] = flex.IntValue(model.APIVersion)
+	modelMap["managed_by"] = *model.ManagedBy
 	return modelMap, nil
 }
 
 func DataSourceIBMAtrackerTargetsCosEndpointToMap(model *atrackerv2.CosEndpoint) (map[string]interface{}, error) {
 	modelMap := make(map[string]interface{})
-	if model.Endpoint != nil {
-		modelMap["endpoint"] = *model.Endpoint
-	}
-	if model.TargetCRN != nil {
-		modelMap["target_crn"] = *model.TargetCRN
-	}
-	if model.Bucket != nil {
-		modelMap["bucket"] = *model.Bucket
-	}
-	if model.ServiceToServiceEnabled != nil {
-		modelMap["service_to_service_enabled"] = *model.ServiceToServiceEnabled
-	}
+	modelMap["endpoint"] = *model.Endpoint
+	modelMap["target_crn"] = *model.TargetCRN
+	modelMap["bucket"] = *model.Bucket
+	modelMap["service_to_service_enabled"] = *model.ServiceToServiceEnabled
 	return modelMap, nil
 }
 
 func DataSourceIBMAtrackerTargetsEventstreamsEndpointToMap(model *atrackerv2.EventstreamsEndpoint) (map[string]interface{}, error) {
 	modelMap := make(map[string]interface{})
-	if model.TargetCRN != nil {
-		modelMap["target_crn"] = *model.TargetCRN
-	}
-	if model.Brokers != nil {
-		modelMap["brokers"] = model.Brokers
-	}
-	if model.Topic != nil {
-		modelMap["topic"] = *model.Topic
-	}
+	modelMap["target_crn"] = *model.TargetCRN
+	modelMap["brokers"] = model.Brokers
+	modelMap["topic"] = *model.Topic
 	if model.APIKey != nil {
-		modelMap["api_key"] = *model.APIKey // pragma: allowlist secret
+		modelMap["api_key"] = *model.APIKey
```

Comment:
Weird that in this function we do not have apikey redacted. Seems like redacted is what we always want, I don't think the state stores apikey or the API response returns apikey.

Which is related to a Teraform bug that has existed for a while where apikey changes do not trigger a TF update, but that's unrelated to what you're doing.